.github/actions/setup-ccache/action.yml

```diff
@@ -0,0 +1,17 @@
+name: Setup ccache
+inputs:
+  extra-cache-key:
+    required: false
+    default: ''
+runs:
+  using: composite
+  steps:
+    - name: Get Date
+      id: get-date
+      run: echo "::set-output name=today::$(/bin/date -u '+%Y%m%d')"
+      shell: bash
+
+    - name: Setup ccache
+      uses: hendrikmuhs/ccache-action@2181be813387616fc2b8dca72d3ff8b912b25f73
+      with:
+        key: ${{ runner.os }}--${{ runner.arch }}--${{ github.workflow }}--${{ steps.get-date.outputs.today }}--${{ inputs.extra-cache-key }}
```

Comment:
So the cache won't be used if a pyx, .c, etc file is modified?

Reply:
The cache will always be used but if a pyx (=> c) file is modified the cache for that particular object will not be used. You can think of ccache as something along the lines of

```
cache_key = compute_hash(preprocess_file(myfile))
if cache_key not in cache:
    cache[cache_key] = run_compiler()
return cache[cache_key]
```

Reply:
Right, could you explain how the hash is being computed here based on a push?

Reply:
Ccache computes it from the preprocessed C source code.

Reply:
All we have to deal with here is syncing the ccache folder to/from GHA cache storage. This is what we do above with the key. We can never have bad cache invalidation in the sense that we use stale compilation results (unless there is a bug in Ccache)